templates/terraform/examples/base_configs/test_file.go.erb

```diff
@@ -53,12 +69,18 @@ func TestAcc<%= test_slug -%>(t *testing.T) {
 
 	resource.Test(t, resource.TestCase{
 		PreCheck:     func() { testAccPreCheck(t) },
+		<% if example_version == 'ga' -%>
 		Providers:    testAccProviders,
+		<% else -%>
+		Providers:    testAccProvidersOiCS,
```

Comment:
This is a bit misleading. The only difference `testAccProvidersOiCS` appears to have is to using the beta provider. What is the OiCS functionality?

Reply:
Encoding the alternate provider set into a variable name was hard so I used the escape hatch of naming it after the only user 🙂 

There's a couple goals we're encoding here;

* We want to use a valid provider version for every resource

* We want our Terraform acceptance tests that verify our OiCS code's correctness

Keeping in mind that `google` in TPGB's tests is actually `google-beta` and we're unable to use TPG without vendoring, the only way we can guarantee that we're using the correct version is to only use a single version and to exclude all the others.

A (purposeful) side effect is that using `google-beta` features without versioning the example properly is that `google-beta` features will end up in `google`, failing a test and ensuring we version examples properly.

Reply:
My preference would be to name something like ProviderBeta. The OICS functionality is tangential and not terrible relevant when running/debugging the test suite. There shouldn't be a need for variables since this only supports ga/beta at this point.

Reply:
> There shouldn't be a need for variables since this only supports ga/beta at this point.

Not sure what you mean by this

Reply:
I just meant you can hard code `testAccProvidersBeta`

Reply:
Also, using a generic name like `ProvidersBeta` could mean other contributors could use that provider set when it's only intended for OiCS-generated tests because of those specific restrictions. How strongly do you feel about naming it something like that?

Reply:
Also to be clear, it's intentional that only beta-exclusive examples use `testAccProvidersOiCS` ; shared examples continue to use `testAccProviders`, so it can't be hardcoded.

Reply:
It sounds like beta only tests should use the beta provider. And until the import is supported we default to the non-beta provider otherwise the tests would fail or need importing disabled. If my summation is correct, the fact that OiCS is also supported as part of test generation isn't relevant. For clarity, I was thinking something like:
```
 diff --git a/templates/terraform/examples/base_configs/test_file.go.erb b/templates/terraform/examples/base_configs/test_file.go.erb
 index d622b594..6b988bbb 100644
 --- a/templates/terraform/examples/base_configs/test_file.go.erb
 +++ b/templates/terraform/examples/base_configs/test_file.go.erb
 @@ -72,7 +72,7 @@ func TestAcc<%= test_slug -%>(t *testing.T) {
         <% if example_version == 'ga' -%>
         Providers:    testAccProviders,
         <% else -%>
 -       Providers:    testAccProvidersOiCS,
 +       Providers:    testAccProvidersBeta,
         <% end -%>
         CheckDestroy: testAccCheck<%= "#{resource_name}" -%>Destroy,
         Steps: []resource.TestStep{
 diff --git a/third_party/terraform/utils/provider_test.go.erb b/third_party/terraform/utils/provider_test.go.erb
 index 844298b3..e911647d 100644
 --- a/third_party/terraform/utils/provider_test.go.erb
 +++ b/third_party/terraform/utils/provider_test.go.erb
 @@ -15,7 +15,7 @@ import (

  var testAccProviders map[string]terraform.ResourceProvider
  <% unless version == 'ga' -%>
 -var testAccProvidersOiCS map[string]terraform.ResourceProvider
 +var testAccProvidersBeta map[string]terraform.ResourceProvider
  <% end -%>
  var testAccProvider *schema.Provider
  var testAccRandomProvider *schema.Provider
 @@ -75,7 +75,7 @@ func init() {
         "random": testAccRandomProvider,
     }

 -   testAccProvidersOiCS = map[string]terraform.ResourceProvider{
 +   testAccProvidersBeta = map[string]terraform.ResourceProvider{
         "google-beta": testAccProvider,
         "random": testAccRandomProvider,
     }
```

Reply:
I think that calling it the beta providers map is misleading, because beta tests should continue to use the standard providers map. It exists exclusively for OiCS, so I've commented clarifying some more with https://github.com/GoogleCloudPlatform/magic-modules/pull/1464/commits/362f4a2aa165b727d2440601e68804083a801634

I don't understand what's hard coded in that example.

Reply:
If this doesn't make sense then no worries. It's not a blocking request 👍🏼